src/shared/telemetry/activation.ts

```diff
@@ -83,13 +105,18 @@ export function isTelemetryEnabled(toolkitSettings: SettingsConfiguration): bool
         return false
     }
 
-    // Current value is expected to be a boolean
-    if (typeof value === 'boolean') {
-        return value
+    //Will throw an error if type is not a boolean
+    try {
+        if (typeof value !== 'boolean') {
+            throw new TypeError(`${value} is not a boolean`)
+        } else {
+            return value
+        }
+    } catch (error) {
```

Comment:
the try/throw/catch ceremony here is overkill. When the type is bool, you can return the value, else log and return true.